src/tokenize_cleanup.cpp

```diff
@@ -1068,3 +1069,86 @@ static void check_template(chunk_t *start)
            (end != NULL) ? get_token_name(end->type) : "<null>");
    set_chunk_type(start, CT_COMPARE);
 } // check_template
+
+
+void cleanup_objc_property(chunk_t *start)
+{
+   chunk_t *open_paren = chunk_get_next_type(start, CT_PAREN_OPEN, start->level);
+
+   if (!open_paren)
+   {
+      LOG_FMT(LTEMPL, "Property is not followed by openning paren\n");
+      return;
+   }
+
+   set_chunk_parent(open_paren, start->type);
+
+   chunk_t *tmp = chunk_get_next_type(start, CT_PAREN_CLOSE, start->level);
+   if (tmp != NULL)
+   {
+      set_chunk_parent(tmp, start->type);
+      tmp = chunk_get_next_ncnl(tmp);
+      if (tmp != NULL)
+      {
+         chunk_flags_set(tmp, PCF_STMT_START | PCF_EXPR_START);
+
+         tmp = chunk_get_next_type(tmp, CT_SEMICOLON, start->level);
+         if (tmp != NULL)
+         {
+            set_chunk_parent(tmp, start->type);
+         }
+      }
+   }
+   mark_selectors_in_property_with_open_paren(open_paren);
+   mark_attributes_in_property_with_open_paren(open_paren);
+}
+
+
+void mark_selectors_in_property_with_open_paren(chunk_t *open_paren)
+{
+   chunk_t *tmp = open_paren;
+
+   assert(open_paren->type == CT_PAREN_OPEN);
+
+   while (tmp && tmp->type != CT_PAREN_CLOSE)
+   {
+      if (tmp->type == CT_WORD &&
+          (memcmp(tmp->str.c_str(), "setter", 6) == 0 ||
+           memcmp(tmp->str.c_str(), "getter", 6) == 0))
+      {
+         tmp = tmp->next;
+         while (tmp && tmp->type != CT_COMMA && tmp->type != CT_PAREN_CLOSE)
+         {
+            if (tmp->type == CT_WORD ||
+                memcmp(tmp->str.c_str(), ":", 1) == 0)
+            {
+               tmp->type = CT_OC_SEL_NAME;
+            }
+            tmp = tmp->next;
+         }
+      }
+      else
+      {
+         tmp = tmp->next;
+      }
+   }
+}
+
+
+void mark_attributes_in_property_with_open_paren(chunk_t *open_paren)
+{
+   chunk_t *tmp = open_paren;
+
+   assert(open_paren->type == CT_PAREN_OPEN);
```

Comment:
You made sure that `open_paren` is not a `nullptr` in `cleanup_objc_property()`, but this function could be used somewhere else in the future where that is not guaranteed anymore. A `nullptr` check for `open_paren` is needed here.